modules/dynamic-operator/controllers/kuberlogicservice-env/env.go

```diff
@@ -117,6 +121,53 @@ func (e *EnvironmentManager) SetupEnv(ctx context.Context) error {
 		}
 	}
 
+	registrySecret := &v1.Secret{
+		ObjectMeta: v12.ObjectMeta{
+			Name:      "docker-registry",
+			Namespace: ns.GetName(),
+		},
+	}
+	if e.cfg.DockerRegistry.Url != "" {
+		dockerConfigJson, err := dockerCredsToJson(
+			e.cfg.DockerRegistry.Url,
+			e.cfg.DockerRegistry.Username,
+			e.cfg.DockerRegistry.Password,
+		)
+		if err != nil {
+			return errors.Wrap(err, "error parsing docker-registry secret")
+		}
+		if _, err := controllerruntime.CreateOrUpdate(ctx, e.Client, registrySecret, func() error {
```

Comment:
You should also set controller reference

Reply:
Done